src/nerf/gun.gleam

```diff
@@ -3,18 +3,34 @@
 
 import gleam/http.{Header}
 import gleam/erlang/charlist.{Charlist}
+import gleam/erlang/atom.{Atom}
 import gleam/dynamic.{Dynamic}
 
 pub external type StreamReference
 
 pub external type ConnectionPid
 
+pub fn open_with_protocols(
```

Comment:
Let's pass an options type to the open function, otherwise we'll end up with a `open_with_*` for every combination of different values that can be passed as options.

We also want to restrict it to a specific `Protocol` type rather than atoms in general as most atoms are not value. `thingy` would be a valid atom, but not a valid protocol.

Reply:
Fixed that, now uses an extra options param